pkg/compose/create.go

```diff
@@ -762,10 +762,6 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, img moby
 						Target:   m.Destination,
 						ReadOnly: !m.RW,
 					}
-					// Avoid mount to be later re-defined
```

Comment:
I'm not sure I remember the detail, but deletion here is expected afaict.
Maybe better create a new `Volumes` before the loop, copy those that need to be kept while iterating, then override `s.Volumes`

Reply:
Yes, either that or a good old `for i := 0; i < len(s.Volumes {`, incrementing i only when not deleting.